lib/web/app/handler.go

```diff
@@ -565,3 +561,21 @@ const (
 	// SubjectCookieName is the name of the application session subject cookie.
 	SubjectCookieName = "__Host-grv_app_session_subject"
 )
+
+func makeAppRedirectURL(r *http.Request, proxyPublicAddr, hostname string) string {
+	// Preserve the app's URL path (URL parts after the app's hostname)
+	// as a query part. Append query that was originally part of the URL.
+	query := fmt.Sprintf("path=%s", url.QueryEscape(r.URL.Path))
+	if len(r.URL.RawQuery) > 0 {
+		query = fmt.Sprintf("%s&%s", query, r.URL.RawQuery)
```

Comment:
Isn't this going to append query with no modifications? What if I'm trying to access my app at `https://app.rafal.teleport.sh/hello?path=1234`?

Reply:
Yeah, it will lose that query param I think.

I'll post more in a sec, I'm looking at the Go APIs for this.